pandas/core/groupby/generic.py

```diff
@@ -840,6 +840,12 @@ def fillna(
         5    NaN
         dtype: float64
         """
+        warnings.warn(
+            'In a future version of pandas, all the parameters except "value" '
+            "of SeriesGroupBy.fillna have to be keyword-only.",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
```

Comment:
This will warn the user on every use of the function. But a user doing `groupby(...).fillna(0)` won't have to change their code, so we shouldn't be warning them. Similarly, it will still warn users who have already changed their code to use keyword args. Instead, use `pandas.util._decorators.deprecate_nonkeyword_arguments` for this purpose. You can find some examples of how it's used throughout the code.

Reply:
isnt this a use case for `@deprecate_nonkeyword_arguments`?